mapper/census.go

```diff
@@ -524,3 +539,56 @@ func generateTruncatePath(path, dimID string, q url.Values) string {
 	}
 	return truncatePath
 }
+
+func getDataLayerJavaScript(analytics map[string]string) (template.JS, error) {
+	jsonStr, err := json.Marshal(analytics)
+	if err != nil {
+		return template.JS(``), err
+	} else {
+		return template.JS(`dataLayer.push(` + string(jsonStr) + `);`), nil
+	}
+}
+
+func getAnalytics(dimensions []model.Dimension) map[string]string {
+	analytics := make(map[string]string, 5)
+	var dimensionIDs []string
+	for _, dimension := range dimensions {
+		if dimension.IsAreaType {
+			analytics["areaType"] = dimension.ID
+			analytics["coverageCount"] = "0"
+		} else if !dimension.IsCoverage {
+			dimensionIDs = append(dimensionIDs, dimension.ID)
+		}
+	}
+	analytics["dimensions"] = strings.Join(dimensionIDs, ",")
+
+	return analytics
+}
+
+func getFilterAnalytics(filterDimensions []sharedModel.FilterDimension) map[string]string {
+	analytics := make(map[string]string, 5)
+	var dimensionIDs []string
+	for _, filterDimension := range filterDimensions {
+		dimension := filterDimension.ModelDimension
+		if dimension.IsAreaType != nil && *dimension.IsAreaType {
+			analytics["areaType"] = dimension.ID
+			analytics["coverageCount"] = strconv.Itoa(len(dimension.Options))
+
+			if len(dimension.Options) > 0 {
+				if len(dimension.Options) <= 4 {
```

Comment:
The int `4` is a bit of a magic number; I had to read the a/c on the ticket to understand why. Please either add a comment or an appropriately named variable